File: solution/java/week05-complexity/exercise2/src/main/java/command/HelpCommand.java
package command;

import java.util.Map;
import java.util.Set;

class HelpCommand implements Command {
    private final Set<String> commandsName;

    public HelpCommand(Set<String> commandsName) {
        this.commandsName = commandsName;
    }

    @Override
    public String executeAndDisplayResult() {
        var result = "Here are all the available commands:\n";
        result += String.join("\n", commandsName);
        return result;
    }
}

File: solution/ts/week05-complexity/exercise2/src/helpCommand.ts
import {Command} from "./command";

export class HelpCommand implements Command {
    private commandsName: Set<string>;

    constructor(commandsName: Set<string>) {
        this.commandsName = commandsName;
    }

    executeAndDisplayResult(): string {
        return `Here are all the available commands:\n${Array.from(this.commandsName).join('\n')}`;
    }
}